nums/core/array/application.py

```diff
@@ -1107,3 +1107,79 @@ def nanmean(self, a: BlockArray, axis=None, keepdims=False, dtype=None):
         if dtype is not None:
             res = res.astype(dtype)
         return res
+
+    def nanvar(self, a: BlockArray, axis=None, ddof=0, keepdims=False, dtype=None):
+        mean = self.nanmean(a, axis=axis, keepdims=True)
+        ss = self.reduce("nansum", (a - mean)**self.two, axis=axis, dtype=dtype, keepdims=keepdims)
+        num_summed = self.sum(~a.ufunc("isnan"), axis=axis, dtype=a.dtype, keepdims=keepdims) - ddof
+        res = ss / num_summed
+        if dtype is not None:
+            res = res.astype(dtype)
+        return res
+
+    def nanstd(self, a: BlockArray, axis=None, ddof=0, keepdims=False, dtype=None):
+        res = self.sqrt(self.nanvar(a, axis, ddof, keepdims))
+        if dtype is not None:
+            res = res.astype(dtype)
+        return res
+
+    def atleast_1d(self, *arys):
+        res = []
+        for ary in arys:
+            if not isinstance(ary, BlockArray):
+                ary = np.array(ary)
+                block_shape = self.compute_block_shape(ary.shape, ary.dtype)
+                ary = self.array(ary, block_shape)
+            if ary.ndim == 0:
+                result = ary.reshape(1)
+            else:
+                result = ary
+            res.append(result)
+        if len(res) == 1:
+            return res[0]
+        else:
+            return res
+
+    def atleast_2d(self, *arys):
+        res = []
+        for ary in arys:
+            if not isinstance(ary, BlockArray):
+                ary = np.array(ary)
+                block_shape = self.compute_block_shape(ary.shape, ary.dtype)
+                ary = self.array(ary, block_shape)
+            if ary.ndim == 0:
+                result = ary.reshape(1, 1)
+
+            # TODO (MWE): Implement this using newaxis when supported
```

Comment:
Until we have clarity on how we want to support the semantics for this case, can we simplify these implementations (from 1d to 3d) so that they all call a single, more general function? If so, can you also describe as a comment why NumPy uses newaxis?

Reply:
Sure.

Reply:
On second thought, generalizing it will just have a lot of messy checks and complicated statements, I think keeping them separate is the best move.